src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/CannotAddOrRemoveVirtualModifier.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+
+#nullable enable
+
+namespace Microsoft.DotNet.ApiCompatibility.Rules
+{
+    /// <summary>
+    /// This class implements a rule to check that the 'virtual' modifier is not added to
+    /// or removed from a member.
+    /// </summary>
+    public class CannotAddOrRemoveVirtualModifier : Rule
+    {
+        public override void Initialize(RuleRunnerContext context) => context.RegisterOnMemberSymbolAction(RunOnMemberSymbol);
+
+        private void RunOnMemberSymbol(ISymbol left, ISymbol right, ITypeSymbol leftContainingType, ITypeSymbol rightContainingType, string leftName, string rightName, IList<CompatDifference> differences)
+        {
+            // Members must exist
+            if (left is null || right is null)
+            {
+                return;
+            }
+            // If the left member is virtual, it must not be sealed or inside a sealed type.
+            if (left.IsVirtual)
+            {
+                // If the right member is not virtual, emit a diagnostic
+                // that the virtual modifier cannot be removed.
+                if (!right.IsVirtual)
+                {
+                    differences.Add(new CompatDifference(
+                    DiagnosticIds.CannotRemoveVirtualModifier, string.Format(
+                        Resources.CannotRemoveVirtualModifier, left), DifferenceType.Removed, right));
+                }
+            }
+            // Otherwise if the left member is not virtual, ensure that
+            // we're either in strict mode, or the left member is not
+            // abstract (since it is legal to change abstract to virtual).
+            else if (Settings.StrictMode || !left.IsAbstract)
```

Comment:
mm I would have expected that an abstract member is ALSO virtual. Is there a case where `left.IsAbstract` is `true` but `left.IsVirtual` is false?

Reply:
I based this off of
https://docs.microsoft.com/en-us/dotnet/core/compatibility/
where it mentions that
✔️ ALLOWED: Changing a member from abstract to virtual

Reply:
Ah I see. Roslyn returns `false` for `IsVirtual` on an abstract member. I think it's just checking for the presence of the keyword, not if they use virtual dispatch underneath the hood.

Reply:
oh interesting, nevermind then.

Reply:
But this begs the question -- when the original issue mentions `virtual`, is it talking about the keyword or if the member is *actually* virtual? Hopefully I didn't misinterpret it. I'll check with Santiago.

Reply:
So it *is* just about the keyword -- so this logic is sound.